hydromt_delft3dfm/dflowfm.py

```diff
@@ -1487,6 +1458,385 @@ def setup_1dboundary(
+        """
+        workflows.find_nearest_branch(
+            branches=branches,
+            geometries=geoms,
+            maxdist=snap_offset,
+        )
+        geoms = geoms.rename(
+            columns={"branch_id": "branchid", "branch_offset": "chainage"}
+        )
+
+        # drop ones non snapped
+        _drop_geoms = geoms["chainage"].isna()
+        if any(_drop_geoms):
+            self.logger.debug(f"Unable to snap to branches: {geoms[_drop_geoms].index}")
+
+        return geoms[~_drop_geoms]
+
+    def setup_1dlateral_from_points(
+        self,
+        laterals_geodataset_fn: str = None,
+        lateral_value: float = 0.0,
+        snap_offset: float = 1.0,
+        branch_type: str = "river",
+    ):
+        """
+        Prepares the 1D lateral discharge from geodataset of point geometries.
+        E.g. '1' m3/s for all lateral locations.
+
+        Use ``laterals_geodataset_fn`` to set the lateral values from a geodataset
+        of point locations.
+        Support also geodataframe of point locations in combination of `lateral_value`.
+
+        Only locations that are snapped to the network of `branch_type`
+        within a max distance defined in ``snap_offset`` are used.
+
+        The discharge can either be a constant using ``lateral_value`` (default) or
+        a timeseries read from ``laterals_geodataset_fn``.
+        If the timeseries has missing values, the constant ``lateral_value`` will be used.
+
+        The timeseries are clipped to the model time based on the model config
+        tstart and tstop entries.
+
+        Adds/Updates model layers:
+            * ** lateral1d_points** forcing: 1D laterals DataArray with points coordinates.
+
+        Parameters
+        ----------
+        laterals_geodataset_fn : str, Path
+            Path or data source name for geospatial point location file.
+            * Required variables if geodataset is provided ['laterals_discharge']
+            NOTE: Require equidistant time series
+        lateral_value : float, optional
+            Constant value to use for all laterals if ``laterals_geodataset_fn`` is a geodataframe,
+            or for filling in missing data.
+            By default 0 [m3/s].
+        snap_offset : float, optional
+            Snapping tolerance to automatically applying boundaries at the correct network nodes.
+            By default 0.1, a small snapping is applied to avoid precision errors.
+        branch_type: str, optional
+            Type of branch to apply laterals on. One of ["river", "pipe"].
+            If None, all branches are used.
+            By defalt None.
+        """
+        self.logger.info(f"Preparing 1D laterals for {branch_type}.")
+        network_by_branchtype = self.staticgeoms[f"{branch_type}s"]
+
+        # 1. read lateral geodataset and snap to network
+        gdf_laterals, da_lat = self._read_forcing_geodataset(
+            laterals_geodataset_fn, "lateral_discharge"
```

Comment:
Do you need the rename here to lateral_discharge versus discharge? I see you rename when adding to forcing to something else?

Reply:
For now not, but maybe good to discuss that together with #114.